src/js/components/mainView/topbar.jsx

```diff
@@ -0,0 +1,28 @@
+define(['lodash', 'react'],
+    function (_, React) {
+        'use strict';
+
+        return React.createClass({
+            displayName: 'TopBar',
+            componentWillMount: function () {
+                if (_.isUndefined(this.props.username) || !this.props.username) {
+                    throw new Error('Failed to create the top bar because the username is missing.');
```

Comment:
This is more of a conceptual comment, i don't think errors should be thrown for bugs.
If username is missing that sound like a bug to me.
but maybe this is something to talk about with everyone